handling-forms/controlled-forms.md

```diff
@@ -0,0 +1,542 @@
+    isPresent: PropTypes.bool,
+    onUpdate: PropTypes.func.isRequired
+};
+
+export default Student;
+```
+<!-- prettier-ignore-end -->
+
+</details>
+
+### Sofia's Plan
+
+Sofia is putting hew new student form into its own component: `NewStudentForm`.
+
+Inside `src/components/NewStudentForm.js`, she begins this component:
+
+<!-- prettier-ignore-start -->
+```js
+const NewStudentForm = () => {
+
+    return (
+        <form>
+            <div>
+                <label htmlFor="fullName">Name:</label>
+                <input name="fullName"/>
+            </div>
+            <div>
+                <label htmlFor="email">Email:</label>
+                <input name="email"/>
+            </div>
+            <input
+                type="submit"
+                value="Add Student"
+            />
+        </form>
+    );
+}
+
+export default NewStudentForm;
+```
+<!-- prettier-ignore-end -->
+
+Each new student should have a name and an email. Her form contains one text input field for name, one for email, and a final submit button.
+
+### !callout-info
+
+## Labels and Names
+
+What's up with the attributes `name` and `for` (mapped in React as `htmlFor`)? These attributes are used by default HTML form behavior. They aren't necessary for this curriculum, but they're a great research topic for creating valid HTML.
+
+### !end-callout
+
+Recall that the steps for creating a controlled form are:
+
+1. Ensure that our text input field's value comes from the component's state.
+1. Ensure that when our text input field changes, it updates the component's state.
+
+Sofia will:
+
+1. Create a new piece of state to hold all of her form data, and make all input values read from state.
+1. Create an event handler for each input field. Each event handler will update state.
+
+### New Piece of State: An Object of `formFields`
+
+Sofia needs to introduce state into the `NewStudentForm` component.
+
+She's made pieces of state that were strings and booleans before, but she's never put an object in state!
+
+To practice that, she'll make one piece of state named `formFields`. It will be an object that will hold `name` and `email` as key-value pairs.
+
+Her vision is that `formFields` will be an easy, nice way to organize the different form fields. By default, a new student's name and email should be blank. The initial state of `formFields` should be:
+
+<!-- prettier-ignore-start -->
+```js
+{
+    name: '',
+    email: ''
+}
+```
+<!-- prettier-ignore-end -->
+
+Therefore, introducing this new piece of state will look like:
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+
+const NewStudentForm = () => {
+
+    const [formFields, setFormFields] = useState({
+        name: '',
+        email: ''
+    });
+```
+<!-- prettier-ignore-end -->
+
+Now she needs to make the input fields read from this state. Since `formFields` is an object, we'll use dot-notation to access the `name` and `email` values.
```

Comment:
Sofia is doing this, and with a bit more context
```suggestion
Now she needs to make the input fields read from this state. Since `formFields` is an object, she can use dot notation to access the `name` and `email` values from the object stored in state.
```